watcher/watcher.go

```diff
@@ -191,5 +206,15 @@ func (watcher Watcher) burnToParams(txid pack.Bytes, amount pack.U256, to pack.S
 	if err != nil {
 		return jsonrpc.ParamsSubmitTx{}, err
 	}
+
+	// Map the v0 burn txhash to v1 txhash so that it is still
+	// queryable
+	v0Hash := v0.BurnTxHash(watcher.selector, pack.NewU256(nonce))
+	watcher.cache.Set(v0Hash.String(), transaction.Hash.String(), 0)
+
+	// Map the selector + burn ref to the v0 hash so that we can return something
+	// to ren-js v1
+	watcher.cache.Set(fmt.Sprintf("%s_%v", watcher.selector, pack.NewU256(nonce).String()), v0Hash.String(), 0)
+
```

Comment:
Not sure this is the right place for this.

Reply:
Where would you suggest? We only ever get the required params when we detect it from the blockchain watcher; the v0 rpc params do not contain the necessary information to construct a valid v1 hash. (also the reason why we ignore the submission in the resolver)

Will add more comments to explain